src/main/java/org/folio/dew/batch/bursarfeesfines/service/BursarTokenFormatter.java

```diff
@@ -278,6 +298,144 @@ public static String formatDataToken(
     }
   }
 
+  private static String formatConstantConditionalAggregatedAccountsToken(
+    BursarExportTokenConstantConditional tokenConstantConditional,
+    AggregatedAccountsByUser aggregatedAccounts
+  ) {
+    List<BursarExportTokenConstantConditionalConditionsInner> conditions = tokenConstantConditional.getConditions();
+
+    for (BursarExportTokenConstantConditionalConditionsInner condition : conditions) {
+      if (
+        BursarFilterAggregateEvaluator.evaluate(
+          aggregatedAccounts,
+          condition.getCondition()
+        )
+      ) {
+        return condition.getValue().getValue();
+      }
+    }
+
+    return tokenConstantConditional.getElse().getValue();
+  }
+
+  private static String formatFeeAmountAggregatedAccountsToken(
+    BursarExportTokenFeeAmount tokenFeeAmount,
+    AggregatedAccountsByUser aggregatedAccounts
+  ) {
+    String result;
+
+    BigDecimal totalAmount = new BigDecimal(0);
+    for (Account account : aggregatedAccounts.getAccounts()) {
+      totalAmount = totalAmount.add(account.getAmount());
+    }
+    if (tokenFeeAmount.getDecimal()) {
+      result = new DecimalFormat("0.00").format(totalAmount);
+    } else {
+      result =
+        totalAmount.multiply(new BigDecimal("100")).setScale(0).toString();
+    }
+
+    return applyLengthControl(result, tokenFeeAmount.getLengthControl());
+  }
+
+  private static String formatUserDataAggregatedAccountsToken(
+    BursarExportTokenUserData tokenUserData,
+    AggregatedAccountsByUser aggregatedAccounts
+  ) {
+    String result;
+    if (
+      tokenUserData.getValue() == BursarExportTokenUserData.ValueEnum.FOLIO_ID
+    ) {
+      result = aggregatedAccounts.getUser().getId();
+    } else if (
+      tokenUserData.getValue() ==
+      BursarExportTokenUserData.ValueEnum.PATRON_GROUP_ID
+    ) {
+      result = aggregatedAccounts.getUser().getPatronGroup();
+    } else {
+      result =
+        String.format(
+          "[unexpected user data token: %s]",
+          tokenUserData.getValue()
+        );
+    }
+
+    return applyLengthControl(result, tokenUserData.getLengthControl());
+  }
+
+  private static String formatUserDataOptionalAggregatedAccountsToken(
```

Comment:
Would it be possible to restructure the user token formatting so that both regular and aggregate formattings could use the same method (e.g. change it to just take the `User` as a parameter, so it can be reused)?

Reply:
I adjusted the user token formatting methods to use `User` as a parameter and use them for both Data Token and Aggregated Token.